### PR TITLE
Implement redux

### PR DESCRIPTION
**Pull request by ichsansandy**
- Understand the concept of state management.
- Use store, actions and reducers in React.
- Connect React and Redux.

### CODE REVIEW

File: src/redux/books/booksSlice.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        item_id: 'item3',
+        title: 'The Selfish Gene',
+        author: 'Richard Dawkins',
+        category: 'Nonfiction',
+      },
+    ],
+  },
+  reducers: {
+    addBook: (state, action) => {
+      state.value.push(action.payload);
```
Comment (Olamarx) [inline src/redux/books/booksSlice.js]: - [ ] I will encourage you not to mutate the state, as it is not a good practice. You can read more on it [here](https://dev.to/rthefounding/redux-never-mutate-state-3jj9), as a result, you can write it like this:
`state.value = [...state.value, action.payload]`

File: src/components/AddNewBookForm.js [hunk 2]
```diff
@@ -1,22 +1,66 @@
...
+            </option>
+            <option value="Fiction">Fiction</option>
+            <option value="Nonfiction">Nonfiction</option>
+            <option value="Biography">Biography</option>
+          </select>
+          <button
+            type="submit"
+            onClick={(e) => {
+              e.preventDefault();
+              if (newBook.author !== '' && newBook.title !== '' && newBook.category !== '') {
```
Comment (Olamarx) [inline src/components/AddNewBookForm.js]: - [OPTIONAL] You can just do if(newBook.author && newBook.title && newBook.category), by that you are checking if they are true, if they are empty, then they are false.

File: src/components/AddNewBookForm.js [hunk 3]
```diff
@@ -1,22 +1,66 @@
...
+                dispatch(addBook({ itemd_id: uuid(), ...newBook }));
+                setNewBook(initialState);
```
Comment (Olamarx) [inline src/components/AddNewBookForm.js:58]: - [ ] I think you should set newbook before dispatching but I have an issue here, whenever I remove your prepopulated data, and add another, regardless of the number of data I add, be it 10, when I delete 1, it deletes everything, I think the id or the delete reducer should be double checked.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @ichsansandy,

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- Readme file professional ✔️
- Add functionality ✔️
- Good use of React Hook ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AbelG101 — APPROVED** [review]
# Hi @ichsansandy 👋, 

**Your project is complete!  There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉 🥳🥇**

## Highlights
- Books are properly added & removed ✔
- State is properly updated ✔
- Actions are properly dispatched ✔
- Correct use of Git Flow ✔
- Well-structured files ✔
- All linter checks are passing ✔

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**